### PR TITLE
feat: add ShapeCast function

### PR DESCRIPTION
**Pull request by zOadT**
This is part of Box2d 2.4.1. There is currently a bug in my implementation, will take a look at it later again.

### CODE REVIEW

File: src/collision/Distance.ts [hunk 1]
```diff
@@ -707,3 +716,164 @@ Distance.Input = DistanceInput;
...
+
+  // Initial simplex
+  const simplex = new Simplex();
+  simplex.m_count = 0;
+
+  // Get simplex vertices as an array.
+  const vertices = simplex.m_v;
+
+  // Get support point in -r direction
+  const indexA = proxyA.getSupport(Rot.mulTVec2(xfA.q, Vec2.neg(r)));
```
Comment (zOadT) [inline src/collision/Distance.ts:777]: BTW: In theory we could also write
```
const indexA = proxyA.getSupport(Rot.mulTVec2(xfA.q, r).neg());
```
Creates a Vec2 object less, but I don't think its worth it?

**Comment by zOadT** [comment]
Think I found the problem, the test is wrong. Will try to fix it tomorrow!

**Comment by zOadT** [comment]
Works now, but the example has to be made a bit more accessible

File: src/collision/Distance.ts [hunk 2]
```diff
@@ -707,3 +716,170 @@ Distance.Input = DistanceInput;
...
+/**
+ * Perform a linear shape cast of shape B moving and shape A fixed. Determines
+ * the hit point, normal, and translation fraction.
+ * @returns true if hit, false if there is no hit or an initial overlap
+ */
+//
+// GJK-raycast
+// Algorithm by Gino van den Bergen.
+// "Smooth Mesh Contacts with GJK" in Game Physics Pearls. 2010
+export const ShapeCast = function(output: ShapeCastOutput, input: ShapeCastInput): boolean {
```
Comment (shakiba) [inline src/collision/Distance.ts:750]: could we put this in its own file?

Comment (zOadT) [reply src/collision/Distance.ts:750]: We would only have to export Simplex from Distance.ts. Would that be ok? Or should I move Simplex into its own file?

Comment (shakiba) [reply src/collision/Distance.ts:750]: Nevermind actually, this is good, thanks

**Review by shakiba — APPROVED** [review]

File: src/collision/Distance.ts (hunk 1 above)
Comment (shakiba) [reply src/collision/Distance.ts]: Yes, it's ok, I'm changing vector functions anyway

**Pull request merged**